src/System.Security.Cryptography.Algorithms/tests/HashAlgorithmTest.cs

```diff
@@ -31,6 +32,177 @@ protected void Verify(Stream input, string output)
+        {
+            using (HashAlgorithm hash = Create())
+            {
+                // TransformBlock + ComputeHash
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                byte[] actual = hash.ComputeHash(block2, 0, block2.Length);
+                Assert.Equal(expected, actual);
+
+                // Verify ComputeHash cleared hash
+                actual = hash.ComputeHash(Array.Empty<byte>(), 0, 0);
+                Assert.Equal(expectedEmpty, actual);
+
+                // Re-hash the same data
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                actual = hash.ComputeHash(block2, 0, block2.Length);
+                Assert.Equal(expected, actual);
+
+                // Verify TransformFinalBlock
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                hash.TransformFinalBlock(block2, 0, block2.Length);
+                Assert.Equal(expected, hash.Hash);
+                Assert.Equal(expected, hash.Hash); // .Hash doesn't clear hash
+
+                // Todo: this will also fail due to the ActiveIssue below in VerifyUseAfterTransformFinalBlock
+                // but commented out so the other tests can run
+                // actual = hash.ComputeHash(Array.Empty<byte>(), 0, 0);
+                // Assert.Equal(expected, actual);
+
+                // TransformBlock + TransformBlock + ComputeHash
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                hash.TransformBlock(block2, 0, block2.Length, null, 0);
+                actual = hash.ComputeHash(Array.Empty<byte>(), 0, 0);
+                Assert.Equal(expected, actual);
+            }
+        }
+
+        [Fact]
+        public void VerifyObjectDisposedException()
+        {
+            HashAlgorithm hash = Create();
+            hash.Dispose();
+            Assert.Throws<ObjectDisposedException>(() => hash.Hash);
+            Assert.Throws<ObjectDisposedException>(() => hash.ComputeHash(Array.Empty<byte>()));
+            Assert.Throws<ObjectDisposedException>(() => hash.ComputeHash(Array.Empty<byte>(), 0, 0));
+            Assert.Throws<ObjectDisposedException>(() => hash.ComputeHash((Stream)null));
+            Assert.Throws<ObjectDisposedException>(() => hash.TransformBlock(Array.Empty<byte>(), 0, 0, null, 0));
+            Assert.Throws<ObjectDisposedException>(() => hash.TransformFinalBlock(Array.Empty<byte>(), 0, 0));
+        }
+
+        [Fact]
+        public void VerifyHashNotYetFinalized()
+        {
+            using (HashAlgorithm hash = Create())
+            {
+                hash.TransformBlock(Array.Empty<byte>(), 0, 0, null, 0);
+                Assert.Throws<CryptographicUnexpectedOperationException>(() => hash.Hash);
+            }
+        }
+
+        [Fact]
+        [ActiveIssue(0)] // Todo: netfx treats HashFinal and Initialize separately, corefx doesn't and Initialize is a no-op
```

Comment:
The ActiveIssue is temporary and for discussion; I do not intend to submit that way. For now I will emulate netfx behavior unless I hear objections otherwise.